client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/site-preview/hooks/use-site-preview-mshot-image-handler.ts

```diff
@@ -0,0 +1,102 @@
+import { useEffect, useRef, useState } from '@wordpress/element';
+import { throttle } from 'lodash';
+
+interface MShotConfig {
+	vpw: number;
+	vph: number;
+	w: number;
+	h: number;
+	screen_height: number;
+	scale: number;
+}
+
+const mShotConfigs: Record< string, MShotConfig > = {
+	desktop: {
+		vpw: 1600,
+		vph: 1600,
+		w: 1600,
+		h: 1624,
+		screen_height: 1600,
+		scale: 2,
+	},
+	tablet: {
+		vpw: 767,
+		vph: 1600,
+		w: 767,
+		h: 1600,
+		screen_height: 1600,
+		scale: 2,
+	},
+	mobile: {
+		vpw: 479,
+		vph: 1200,
+		w: 479,
+		h: 1200,
+		screen_height: 1200,
+		scale: 2,
+	},
+};
+
+export const useSitePreviewMShotImageHandler = () => {
+	const [ mShotsOption, setMShotsOption ] = useState< MShotConfig | undefined >( undefined );
+	const [ currentSegment, setCurrentSegment ] = useState( '' );
+
+	const getSegment = ( width: number ) => {
+		switch ( true ) {
+			case width >= 1024:
+				return 'desktop';
+			case width >= 600:
+				return 'tablet';
+			default:
+				return 'mobile';
+		}
+	};
+
+	const previewRef = useRef< HTMLDivElement >( null );
+
+	const updateDimensions = ( previewRef: React.RefObject< HTMLDivElement > ) => {
+		if ( previewRef.current ) {
+			const { offsetWidth } = previewRef.current;
+			const width = Math.min( offsetWidth, 1920 );
+
+			const newSegment = getSegment( width );
+
+			if ( currentSegment === newSegment ) {
+				return;
+			}
+
+			setMShotsOption( mShotConfigs[ newSegment ] );
+			setCurrentSegment( newSegment );
+		}
+	};
+
+	useEffect( () => {
+		updateDimensions( previewRef );
```

Comment:
I think we can skip this if the `previewRef` is not added to an element, right? For the case that we pre-fetch it.

Reply:
Sorry I'm not sure I understand it correctly. Is there a case when it's not added to an element?

Reply:
I mean in the identify step that we use to pre-fetch. In this case, I suspect we don't need to run this, right?

Reply:
Oh got you, here it is https://github.com/Automattic/wp-calypso/pull/92355/commits/b9446dbf96a97dda16b7bbf727d9633f66e1c5e1 . Thank you!